File: backend/studio515/src/domain/product.ts
import { prop, arrayProp , Ref, Typegoose } from '@typegoose/typegoose';
import { ProductImage } from './product_image';
import { Field, ID, ObjectType } from 'type-graphql';

@ObjectType()
export class Product extends Typegoose {
    @Field(type => ID)
    @prop()
    id: number;

    @Field({ nullable: true })
    @prop()
    name?: string;

    @Field({ nullable: true })
    @prop()
    prod_info?: string;

    @Field(type => [ProductImage])
    @arrayProp ({ items: ProductImage })
    images?: Ref<ProductImage>[];


    @Field({ nullable: true })
    @prop()
    video?: string;

    @Field({ nullable: true })
    @prop()
    video_image?: string;


    @Field({ nullable: true })
    @prop()
    about?: string;

    @Field({ nullable: true })
    @prop()
    cat1?: number; //一级页面分类

    @Field({ nullable: true })
    @prop()
    category_id?: number; //图片分类

    @Field({ nullable: true })
    @prop()
    created_at?: Date;

    @Field({ nullable: true })
    @prop()
    sort_id?: number;
}

File: flutter/lib/model/product.dart
import 'package:flutter/material.dart';
import 'package:studio515/model/product_image.dart';

class Product {
//  String _id;
  int    id;
  String name;
  String created_at;
  int sort_id;
  String about;
  int cat1;
  int category_id;

  List<ProductImage> images;
  
  Product.fromJson(Map data) {
//    _id            = data['_id'];
    id             = data['id'];
    name           = data['name'];
    created_at     = data['created_at'];
    sort_id        = data['sort_id'];
    about          = data['about'];
    cat1           = data['cat1'];
    category_id    = data['category_id'];
    images         = new List<ProductImage>();
    if (data['images'] != null) {
      data['images'].forEach((data) {
        var im = ProductImage.fromJson(data);
        if(im != null){
          images.add(im);
        }
      });
    }
  }
}
